aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -79,7 +79,7 @@ int64_t SparseTensorImpl::storage_offset() const {
   AT_ERROR("sparse tensors do not have storage");
 }
 void SparseTensorImpl::set_indices_and_values_unsafe(const Tensor& indices, const Tensor& values) {
-  AT_CHECK(allow_tensor_metadata_change(), "set_indices_and_values_unsafe is not allowed on Tensor created from .data or .detach()");
+  AT_CHECK(allow_tensor_metadata_change(), "set_indices_and_values_unsafe " + FLAGS_err_msg_for_changing_metadata_on_derived_tensor);
```

Comment:
FYI you don't need to do a `std::string` append; `AT_CHECK` is variadic and will concat each of the arguments together.